lib/export/jdl_exporter.js

```diff
@@ -18,11 +18,155 @@
+
+/**
+ * Indicate if an unary option should be written in the jdl file
+ * @param key
+ */
+function unaryOptionsExists(key) {
+  return !excludeUnaryOptions.includes(key) && unaryOptions.exists(key);
+}
+
+const excludeOptions = ['jhipsterVersion', 'packageFolder', 'otherModules'];
+
+/**
+ * Indicate if an application option should be written in the jdl file
+ * @param key option name
+ */
+function applicationOptionsExists(key) {
+  return !excludeOptions.includes(key) && applicationOptions.exists(key);
+}
+
+/**
+ * Remove unwanted application options for the jdl file
+ * @param application the application
+ */
+function filterApplicationOptions(application) {
+  if (application.config === undefined) {
+    throw new Error(`'${application.toString()}' must contain config field.`);
+  }
+  const config = application.config;
+  Object.keys(config).forEach(key => {
+    if (!applicationOptionsExists(key)) {
+      delete config[key];
+    } else if ((key === 'entitySuffix' || key === 'dtoSuffix') && config[key] === '') {
+      config[key] = false;
+    }
+  });
+  return application;
+}
+
+/**
+ * load the application configuration and the associated entities from a Jhipster project
+ * contained in a folder
+ * @param dir path to the directory where it should contains the .yo-rc.json
+ */
+function loadApplicationAndEntities(dir) {
+  if (!dir) {
+    dir = '.';
+  }
+
+  if (!FileUtils.doesFileExist(path.join(dir, '.yo-rc.json'))) {
+    throw new Error('The directory does not contain a .yo-rc.json');
+  }
+  const jsonapplication = JSONFileReader.readEntityJSON(path.join(dir, '.yo-rc.json'));
+  let jdlapplication = JSONToJdlApplicationConverter.convertApplicationsToJDL({
+    applications: [jsonapplication]
+  }).applications;
+  jdlapplication = jdlapplication[Object.keys(jdlapplication)[0]];
+  let jdlentities = new JDLObject();
+
+  if (FileUtils.doesDirectoryExist(path.join(dir, '.jhipster'))) {
+    jdlentities = JSONReader.parseFromDir(dir);
+  }
+
+  addEntitiesToJdlApplication(jdlentities, jdlapplication);
+  filterApplicationOptions(jdlapplication);
+  return {
+    JDLApplication: jdlapplication,
+    JDLEntities: jdlentities
+  };
+}
+
+/**
+ * Add the entity names used by the application
+ * @param entitiesjdl entities
+ * @param applicationjdl application
+ */
+function addEntitiesToJdlApplication(entitiesjdl, applicationjdl) {
+  const entities = entitiesjdl.entities;
+  const entitynames = new Set();
+  Object.keys(entities).forEach(key => {
+    entitynames.add(entities[key].name);
+  });
+  applicationjdl.entityNames = entitynames;
+  return applicationjdl;
+}
+
+/**
+ * load a jhipster configuration with the entities into a JDLObject
+ * @param dir root folder
+ */
+function loadJhipsterConfigAndEntities(dir) {
+  const jdl = new JDLObject();
+  const apps = [];
+
+  if (FileUtils.doesFileExist(path.join(dir, '.yo-rc.json'))) {
+    apps.push(loadApplicationAndEntities(dir));
```

Comment:
If you call the `loadApplicationAndEntities`, you should put it after the `loadJhipsterConfigAndEntities`